scripts/delete-unused-users.py

```diff
@@ -36,15 +41,24 @@ async def main():
                 except:
                     print(user['last_activity'])
                     raise
-                if last_activity and datetime.now().astimezone() - last_activity < timedelta(hours=24) or user['server'] is not None:
+                delta_bool = datetime.now().astimezone() - last_activity < timedelta(hours=24)
+                print(f"user: {user['name']}")
+                print(f"last login: {last_activity}")
+                print(f"24hrs since last login: {delta_bool}")
+                print(f"server: {user['server']}")
+                if (last_activity and delta_bool) or (user['server'] is not None):
```

Comment:
@ryanlovett @gmerritt could you look at the original `if` statement (like 39 of original file) and confirm i unpacked this properly?  my test runs lead me to believe i did, but another set of eyeballs would make me happier.

Reply:
If `last_activity` is ever something other than a datetime object (if `parse` returns None for example), then the `delta_bool` assignment will fail. 

I might also rename `delta_bool` to be `time_since_last_activity` or some such to make it a little quicker for human eyes to parse.

Reply:
i added some type checking logic, but looking at the docs for `dateutil.parser.parse` it should return a `datetime.datetime` object or raise an exception.  if it raises an exception, the script won't continue and we won't hit the condition you were worried about.